var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -62,53 +73,82 @@ class Zfp(CMakePackage, CudaPackage):
         "use smaller for finer rate granularity. "
         "Use 8 for H5Z-ZFP filter.",
     )
+
     variant(
-        "strided", default=False, description="Enable strided access for progressive zfp streams"
+        "daz",
+        default=False,
+        description="Denormals are zero: "
+        "Treat denormal-only blocks as containing "
+        "all zeroes",
     )
-    variant("aligned", default=False, description="Enable aligned memory allocation")
-    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     variant(
         "fasthash",
         default=False,
         description="Use a faster but more collision prone hash function",
     )
+
     variant("profile", default=False, description="Count cache misses")
 
+    variant(
+        "round",
+        default="ZFP_ROUND_NEVER",
+        values=("ZFP_ROUND_NEVER", "ZFP_ROUND_FIRST", "ZFP_ROUND_LAST"),
+        multi=False,
+        description="EXPERIMENTAL: Set coefficient rounding method",
+    )
+
+    variant(
+        "strided", default=False, description="Enable strided access for progressive zfp streams"
+    )
+
+    variant(
+        "tight_error",
+        default=False,
+        description="EXPERIMENTAL: Enable rounding-based error reduction",
+    )
+
+    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     # Conflicts
     conflicts("+c", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+python", when="@:0.5.4", msg="+python requires zfp 0.5.5 or later")
-    conflicts("+fortran", when="@:0.5.4", msg="+fortran requires zfp 0.5.5 or later")
-    conflicts("+openmp", when="@:0.5.2", msg="+openmp requires zfp 0.5.3 or later")
-    conflicts("+cuda", when="@:0.5.3", msg="+cuda requires zfp 0.5.4 or later")
+    conflicts("+daz", when="@:0.5.5", msg="+daz required zfp 1.0.0 or later")
+    conflicts("+python", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+fortran", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+openmp", when="@:0.5.2", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+cuda", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
     conflicts("+fasthash", when="@:0.5.1", msg="+fasthash requires zfp 0.5.2 or later")
     conflicts("+profile", when="@:0.5.1", msg="+profile requires zfp 0.5.2 or later")
+    conflicts("round", when="@:0.5.5", msg="+round requires zfp 1.0.0 or later")
```

Comment:
You'll also want one conflict for each of the possible values, e.g.

```suggestion
    conflicts("round=ZFP_ROUND_FIRST", when="@:0.5.5", msg="+round requires zfp 1.0.0 or later")
```

Otherwise, the conflict won't trigger.

Reply:
Thanks for catching this. It has been a little tricky figuring out some of the finer details of non-boolean variants and associated conflicts.